packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -137,6 +138,16 @@ export class CanvasManager {
     let lastSnapshotTime = 0;
     let rafId: number;
 
+    const getCanvas = (): HTMLCanvasElement[] => {
+      const matchedCanvas: HTMLCanvasElement[] = [];
+      win.document.querySelectorAll('canvas').forEach(canvas => {
+        if (!isBlocked(canvas, blockClass, false)) {
```

Comment:
In this case you need to check if the ancestors are blocked too
```suggestion
        if (!isBlocked(canvas, blockClass, true)) {
```